src/main/java/picard/analysis/TheoreticalSensitivity.java

```diff
@@ -204,4 +204,174 @@ public int draw() {
+     * @param randomSeed random number seed to use for random number generator
+     * @return
+     */
+    public static double sensitivityAtConstantDepth(final int depth, final double[] qualityDistribution, final double logOddsThreshold, final int sampleSize, final double alleleFraction, final int randomSeed) {
+        final RouletteWheel qualityRW = new RouletteWheel(trimDistribution(qualityDistribution));
+        final Random uniformRNG = new Random(randomSeed);
+
+        int altDepth = 0;
+        int calledVariants = 0;
+        for (int k = 0; k < sampleSize; k++) {
+            altDepth = binomialDraw(depth, alleleFraction, uniformRNG);
+
+            int sumOfQualities = 0;
+            for (int i = 0; i < altDepth; i++) {
+                sumOfQualities += qualityRW.draw();
+            }
+            if (isCalled(depth, altDepth, (double) sumOfQualities / (double) altDepth, alleleFraction, logOddsThreshold)) {
+                calledVariants++;
+            }
+        }
+        return (double) calledVariants / sampleSize;
+    }
+
+    /**
+     * Calculates the theoretical sensitivity with a given Phred-scaled quality score distribution at a constant
+     * depth.
+     * @param depth Depth to compute sensitivity at
+     * @param qualityDistribution Phred-scaled quality score distribution
+     * @param logOddsThreshold Log odds threshold necessary to exceed for variant to be called
+     * @param sampleSize the total number of simulations to run
+     * @param alleleFraction the allele fraction to evaluate sensitivity at
+     * @return
+     */
+    public static double sensitivityAtConstantDepth(final int depth, final double[] qualityDistribution, final double logOddsThreshold, final int sampleSize, final double alleleFraction) {
+        return sensitivityAtConstantDepth(depth, qualityDistribution, logOddsThreshold, sampleSize, alleleFraction, randomSeed);
+    }
+
+    /**
+     * Calculates the theoretical sensitivity with a given Phred-scaled quality score distribution and depth
+     * distribution.
+     * @param depthDistribution Depth distribution to compute theoretical sensitivity over
+     * @param qualityDistribution Phred-scaled quality score distribution
+     * @param sampleSize the total number of simulations to run
+     * @param logOddsThreshold Log odds threshold necessary to exceed for variant to be called
+     * @param alleleFraction the allele fraction to evaluate sensitivity at
+     * @return
+     */
+    public static double theoreticalSensitivity(final double[] depthDistribution, final double[] qualityDistribution,
+                                                final int sampleSize, final double logOddsThreshold, final double alleleFraction) {
+        if (alleleFraction > 1.0 || alleleFraction < 0.0) {
+            throw new PicardException("Allele fractions must be between 0 and 1.");
+        }
+        double sensitivity = 0.0;
+        for (int k = 0; k < depthDistribution.length; k++) {
```

Comment:
Given that you want to go to high depth, it would make sense to bin the depth distribution in a smart manner (that gives higher fidelity for small depths) and iterate over that binned distribution.